File: hw/arias/U3/HW37StrategyPattern/SortApp/src/main/java/ec/edu/espe/sortapp/model/BubbleSort.java


package ec.edu.espe.sortapp.model;

/**
 *
 * @author Alan Arias
 */
public class BubbleSort implements SortingStrategy {
    public int[] sort(int data[]) {
        int n = data.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (data[j] > data[j + 1]) {
                    int temp = data[j];
                    data[j] = data[j + 1];
                    data[j + 1] = temp;
                }
            }
        }
        return data;
    }
}

File: EXAM/pazmino/u3/PazmiñoStrategy/ec/edu/espe/pazmiñostrategy/model/bubble_sort.py
class BubbleSort:
    
    def sort(self, data):
        if data is None or len(data) < 2:
            return data

        size = len(data)
        for i in range(size - 1):
            swapped = False
            for j in range(size - i - 1):
                if data[j] > data[j + 1]:
                    data[j], data[j + 1] = data[j + 1], data[j]
                    swapped = True
            if not swapped:
                break

        return data
